client-projects/the-lumery/filterButtons.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  FILTER_BUTTON_LIST_SELECTOR,
+  FILTER_RESET_BUTTON_SELECTOR,
+} from "./selectors";
+
+export function setUpFilterButtons() {
+  const resetButton: HTMLElement | null = document.querySelector(
+    FILTER_RESET_BUTTON_SELECTOR,
+  );
+  if (!resetButton) {
+    throw new Error(
+      `Reset button element not found: ${FILTER_RESET_BUTTON_SELECTOR}`,
+    );
+  }
+  const filterButtonList: HTMLElement | null = document.querySelector(
+    FILTER_BUTTON_LIST_SELECTOR,
+  );
+  if (!filterButtonList) {
+    throw new Error(
+      `Filter button list element not found: ${FILTER_BUTTON_LIST_SELECTOR}`,
+    );
+  }
+
+  filterButtonList.insertBefore(
+    resetButton,
+    filterButtonList.firstElementChild,
+  );
+
+  resetButton.style.opacity = "1";
+
+  filterButtonList.animate([{ opacity: 0 }, { opacity: 1 }], {
+    duration: 400,
+    delay: 900, // allow fonts to load and the header animation to finish
```

Comment:
How much of this 900 is waiting for fonts (vs waiting for the header animation)? Don't love the magic number-ness of it, but I guess that ensuring that the font gets loaded is both tricky and unnecessarily blocking so 🤷

Reply:
Fonts get loaded at around 500/600ms on my machine and connection. I _could_ do some check for the fonts before fading in the buttons, but I don't want to have new stuff appearing on screen while the header is animating, so I extended the delay to fade in at the end of the animation